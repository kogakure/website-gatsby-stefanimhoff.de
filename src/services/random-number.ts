export const getRandomNumber = (max: number) =>
  String(Math.floor(Math.random() * Math.floor(max)));
