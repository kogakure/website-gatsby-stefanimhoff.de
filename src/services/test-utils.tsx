import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';

const customRender = (
  node: React.ReactElement,
  options?: RenderOptions
): RenderResult => {
  const rendered = render(
    <EmojifyProvider>
      <LocalizedDateProvider>{node}</LocalizedDateProvider>
    </EmojifyProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (ui: React.ReactElement, rerenderOptions?: RenderOptions) =>
      customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
