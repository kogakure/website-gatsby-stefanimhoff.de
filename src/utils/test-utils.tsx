import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';
import { ThemeType } from '../typings/Theme';
import { lightTheme } from '../themes';

interface CustomTheme {
  theme: ThemeType;
}

type RenderOptionsWithTheme = RenderOptions & CustomTheme;

const customRender = (
  node: React.ReactElement,
  options: RenderOptionsWithTheme = { theme: lightTheme }
): RenderResult => {
  const { theme } = options;
  const rendered = render(
    <ThemeProvider theme={theme}>
      <EmojifyProvider>
        <LocalizedDateProvider>{node}</LocalizedDateProvider>
      </EmojifyProvider>
    </ThemeProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (
      ui: React.ReactElement,
      rerenderOptions?: RenderOptionsWithTheme
    ) => customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
