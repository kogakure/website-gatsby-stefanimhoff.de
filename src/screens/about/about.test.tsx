import React from 'react';
import * as Gatsby from 'gatsby';
import { axe } from 'jest-axe';

import { render } from '../../services/test-utils';

import {
  PhotoContainer,
  PortraitPhoto,
  SectionAbout,
  SectionAboutWebsite,
  SectionConnect,
  SectionContact,
  SocialMedia,
  TitleAbout,
} from '.';

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');

useStaticQuery.mockImplementation(() => ({
  placeholderImage: {
    childImageSharp: {
      fluid: {
        aspectRatio: 0.66,
        base64: 'data:image/jpeg;base64,XYZ',
        sizes: '(max-width: 800px) 100vw, 800px',
        src: '/static/image.jpg',
        srcSet: `/static/image.jpg 200w,
                 /static/image.jpg 400w,
                 /static/image.jpg 800w,
                 /static/image.jpg 1200w,
                 /static/image.jpg 1600w,
                 /static/image.jpg 2000w`,
      },
    },
  },
}));

describe('About', () => {
  describe('TitleAbout', () => {
    test('renders correctly', () => {
      const { container } = render(<TitleAbout>About</TitleAbout>);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('is accessible', async () => {
      const { container } = render(<TitleAbout>About</TitleAbout>);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });

  describe('PhotoContainer', () => {
    test('renders correctly', () => {
      const { container } = render(<PhotoContainer>Photo</PhotoContainer>);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('is accessible', async () => {
      const { container } = render(<PhotoContainer>Photo</PhotoContainer>);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });

  describe('SectionAbout', () => {
    test('renders correctly', () => {
      const { container } = render(<SectionAbout>About</SectionAbout>);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('is accessible', async () => {
      const { container } = render(<SectionAbout>About</SectionAbout>);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });

  describe('SectionAboutWebsite', () => {
    test('renders correctly', () => {
      const { container } = render(
        <SectionAboutWebsite>About Website</SectionAboutWebsite>
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    test('is accessible', async () => {
      const { container } = render(
        <SectionAboutWebsite>About Website</SectionAboutWebsite>
      );
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });

  describe('SectionContact', () => {
    test('renders correctly', () => {
      const { container } = render(<SectionContact>Contact</SectionContact>);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('is accessible', async () => {
      const { container } = render(<SectionContact>Contact</SectionContact>);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });

  describe('SectionConnect', () => {
    test('renders correctly', () => {
      const { container } = render(<SectionConnect>Connect</SectionConnect>);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('is accessible', async () => {
      const { container } = render(<SectionConnect>Connect</SectionConnect>);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });

  describe('SocialMedia', () => {
    test('renders correctly', () => {
      const { container } = render(<SocialMedia>Social Media</SocialMedia>);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('is accessible', async () => {
      const { container } = render(
        <SocialMedia>
          <li>Social Media</li>
        </SocialMedia>
      );
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });

  describe('PortraitPhoto', () => {
    test('renders correctly', () => {
      const { container } = render(<PortraitPhoto />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
