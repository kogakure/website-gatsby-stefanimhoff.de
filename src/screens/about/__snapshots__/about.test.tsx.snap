// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About PhotoContainer renders correctly 1`] = `
.c0 {
  border-bottom-width: 1px;
  border-color: var(--colorForegroundSoft);
  border-left-width: 0;
  border-right-width: 1px;
  border-style: solid;
  border-top-width: 1px;
  grid-column: 1 / 12;
  grid-row: 2;
  height: auto;
  position: relative;
}

@media (min-width:768px) {
  .c0 {
    border-left-width: 1px;
    grid-column: 4 / 10;
  }
}

@media (min-width:1200px) {
  .c0 {
    grid-column: 7 / 10;
  }
}

<div
  class="c0"
>
  Photo
</div>
`;

exports[`About PortraitPhoto renders correctly 1`] = `
<div
  class=" gatsby-image-wrapper"
  style="position: relative; overflow: hidden;"
>
  <div
    aria-hidden="true"
    style="width: 100%; padding-bottom: 151.5151515151515%;"
  />
  <img
    alt=""
    aria-hidden="true"
    src="data:image/jpeg;base64,XYZ"
    style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
  />
  <picture>
    <source
      sizes="(max-width: 800px) 100vw, 800px"
      srcset="/static/image.jpg 200w,
                 /static/image.jpg 400w,
                 /static/image.jpg 800w,
                 /static/image.jpg 1200w,
                 /static/image.jpg 1600w,
                 /static/image.jpg 2000w"
    />
    <img
      alt=""
      loading="lazy"
      sizes="(max-width: 800px) 100vw, 800px"
      src="/static/image.jpg"
      srcset="/static/image.jpg 200w,
                 /static/image.jpg 400w,
                 /static/image.jpg 800w,
                 /static/image.jpg 1200w,
                 /static/image.jpg 1600w,
                 /static/image.jpg 2000w"
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
    />
  </picture>
  <noscript>
    &lt;picture&gt;&lt;source srcset="/static/image.jpg 200w,
                 /static/image.jpg 400w,
                 /static/image.jpg 800w,
                 /static/image.jpg 1200w,
                 /static/image.jpg 1600w,
                 /static/image.jpg 2000w" sizes="(max-width: 800px) 100vw, 800px" /&gt;&lt;img loading="lazy" sizes="(max-width: 800px) 100vw, 800px" srcset="/static/image.jpg 200w,
                 /static/image.jpg 400w,
                 /static/image.jpg 800w,
                 /static/image.jpg 1200w,
                 /static/image.jpg 1600w,
                 /static/image.jpg 2000w" src="/static/image.jpg" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
  </noscript>
</div>
`;

exports[`About SectionAbout renders correctly 1`] = `
.c0 {
  grid-column: 2 / -2;
  grid-row: 3;
}

@media (min-width:768px) {
  .c0 {
    grid-column: 6 / 14;
  }
}

@media (min-width:1200px) {
  .c0 {
    grid-column: 8 / 13;
  }
}

<section
  class="c0"
>
  About
</section>
`;

exports[`About SectionAboutWebsite renders correctly 1`] = `
.c0 {
  grid-column: 2 / -2;
  grid-row: 4;
}

@media (min-width:768px) {
  .c0 {
    grid-column: 5 / 13;
  }
}

@media (min-width:1200px) {
  .c0 {
    grid-column: 7 / 12;
  }
}

<section
  class="c0"
>
  About Website
</section>
`;

exports[`About SectionConnect renders correctly 1`] = `
.c0 {
  grid-column: 2 / -2;
  grid-row: 6;
}

@media (min-width:768px) {
  .c0 {
    grid-column: 8 / -3;
    grid-row: 5;
  }
}

@media (min-width:1200px) {
  .c0 {
    grid-column: 9 / -5;
  }
}

<section
  class="c0"
>
  Connect
</section>
`;

exports[`About SectionContact renders correctly 1`] = `
.c0 {
  grid-column: 2 / -2;
  grid-row: 5;
}

@media (min-width:768px) {
  .c0 {
    grid-column: 3 / 8;
  }
}

@media (min-width:1200px) {
  .c0 {
    grid-column: 5 / 8;
  }
}

<section
  class="c0"
>
  Contact
</section>
`;

exports[`About SocialMedia renders correctly 1`] = `
.c0 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
  list-style-type: circle;
  margin-bottom: clamp(2.074rem,4.44vw,2.488rem);
  padding-left: 1.2em;
}

li .c0 {
  padding-left: 2rem;
}

.c0.contains-task-list {
  list-style-type: none;
}

.c1 {
  -webkit-column-count: 1;
  column-count: 1;
  margin: 0;
}

@media (min-width:768px) {
  .c0 {
    padding-left: 0;
  }
}

@supports (-webkit-appearance:none) or (-moz-appearance:none) or (appearance:none) {
  .c0 input[type='checkbox'] {
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    background: transparent;
    border-color: var(--colorForegroundSoft);
    border-radius: 25%;
    border-style: solid;
    border-width: 0.1em;
    cursor: pointer;
    display: inline-block;
    height: 1em;
    margin: 0;
    outline: none;
    position: relative;
    top: 0.3em;
    vertical-align: top;
    width: 1em;
  }

  .c0 input[type='checkbox']:after {
    border-bottom-width: 0.15em;
    border-color: var(--colorForeground);
    border-left-color: transparent;
    border-left-width: 0;
    border-right-width: 0.15em;
    border-style: solid;
    border-top-color: transparent;
    border-top-width: 0;
    content: '';
    display: block;
    height: 0.5em;
    left: 0.25em;
    opacity: var(--opacity,0);
    position: absolute;
    top: 0.1em;
    -webkit-transform: rotate(var(--rotate,0deg));
    -ms-transform: rotate(var(--rotate,0deg));
    transform: rotate(var(--rotate,0deg));
    width: 0.3em;
  }

  .c0 input[type='checkbox']:checked {
    --rotate: 43deg;
    --opacity: 1;
  }

  .c0 input[type='checkbox']:disabled {
    cursor: not-allowed;
  }
}

@media (min-width:300px) {
  .c1 {
    -webkit-column-count: 2;
    column-count: 2;
    grid-column: 9 / -4;
  }
}

@media (min-width:400px) {
  .c1 {
    -webkit-column-count: 3;
    column-count: 3;
  }
}

<ul
  class="c0 c1"
>
  Social Media
</ul>
`;

exports[`About TitleAbout renders correctly 1`] = `
.c0 {
  font-size: clamp(2.25rem,4.7vw,5.30rem);
  font-weight: 900;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1.0;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  margin-top: 0;
}

.c0 .autolink {
  background: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: var(--colorForeground);
  margin-left: clamp( 0.335rem,0.72vw,0.402rem);
  opacity: 0;
  vertical-align: middle;
}

.c0:hover .autolink,
.c0:focus .autolink {
  opacity: 0.5;
}

.c1 {
  grid-column: 2 / -2;
  grid-row: 1;
  margin: 0;
}

@media screen and (prefers-reduced-motion:no-preference) {
  .c0 .autolink {
    -webkit-transition-duration: 500ms;
    transition-duration: 500ms;
    -webkit-transition-property: opacity;
    transition-property: opacity;
    -webkit-transition-timing-function: ease-in-out;
    transition-timing-function: ease-in-out;
  }
}

@media (min-width:768px) {
  .c1 {
    grid-column: 5 / -7;
  }
}

@media (min-width:1200px) {
  .c1 {
    grid-column: 8 / -7;
  }
}

<h1
  class="c0 c1"
>
  About
</h1>
`;
